common/setup.c

```diff
@@ -44,12 +45,85 @@
 #include <drivers/serial.h>
 #include <slab.h>
 
-bool opt_debug;
+bool opt_debug = 0;
+bool_cmd("debug", opt_debug);
 
 io_port_t com_ports[2];
 
 const char *kernel_cmdline;
 
+static int parse_bool(const char *s) {
+    if (!strcmp("no", s) || !strcmp("off", s) || !strcmp("false", s) ||
+        !strcmp("disable", s) || !strcmp("0", s))
+        return 0;
+
+    if (!strcmp("yes", s) || !strcmp("on", s) || !strcmp("true", s) ||
+        !strcmp("enable", s) || !strcmp("1", s))
+        return 1;
+
+    return -1;
+}
+
+static inline const char *string_trim_whitspace(const char *s) {
+    while(isspace(*s))
+        s++;
+
+    return s;
+}
+
+#define OPT_MAX_SIZE 256
+
+void cmdline_parse(const char *cmdline) {
+    char opt[OPT_MAX_SIZE], *optval, *optkey, *q;
```

Comment:
I would place the `opt[]` into `.bss.init` by `static __bss_init char opt[PAGE_SIZE]`.. and I would use a single 4K for the size.